python/ray/tune/trial_runner.py

```diff
@@ -281,15 +230,15 @@ def _process_events(self):
             trial.update_last_result(
                 result, terminate=(decision == TrialScheduler.STOP))
 
+            self.trial_executor.on_scheduler_decision(trial, decision=decision)
```

Comment:
One thing that isn't clear to me here is what exactly is in scope and out of scope of this hook. The TrialRunner already calling the executor based off the decision, so what should this decision hook do?

Reply:
In RayTrialExecutor, if decision==CONTINUE, ray will call train.remote() function that starts another train iterator.
In addition, hook have  some  other purposes, such as debugging and extending functionality.

Reply:
If we don't need on_scheduler_decision, Maybe we need to add a function like start_iterator or continue_trial inside the TrialExecutor?